src/cursor/cur_index.c

```diff
@@ -347,6 +354,46 @@ __curindex_search_near(WT_CURSOR *cursor, int *exact)
     API_END_RET(session, ret);
 }
 
+/*
+ * __increment_bound_array --
+ *     Increment the given buffer by one bit. Note: This function can potentially extend the buffer
+ *     by one byte, if there is not enough space to fit the one bit.
+ */
+static inline int
+__increment_bound_array(WT_SESSION_IMPL *session, WT_ITEM *user_item)
+{
+    size_t usz;
+    uint8_t *userp;
+    int i;
+
+    usz = user_item->size;
+    userp = (uint8_t *)user_item->data;
+
+    if (usz <= 0)
+        return (0);
+
+    /*
+     * First loop through all max values on the buffer from the end. This is to find the appropriate
+     * position to increment add one to the byte. If we encounter a max value, make it zero due to
+     * the addition of one bit.
+     */
+    for (i = usz - 1; i >= 0 && userp[i] == UINT8_MAX; --i)
+        userp[i] = 0;
+
+    /*
+     * If all of the buffer are max values, extend the buffer by one and add one to the start of the
+     * buffer. Otherwise we have found a position to increment inside the buffer.
+     */
+    if (i < 0) {
+        WT_RET(__wt_buf_extend(session, user_item, user_item->size + 1));
+        userp = (uint8_t *)user_item->data;
+        userp[0] = 1;
```

Comment:
I think this needs to be a little different.  If you started with 0xffff, then you'll you'll currently have a buffer with 0x0000, and you extend it to get 0x000000.  But what you really want is a value that compares bigger (with memcmp) to 0xffff, so I think 0xffff00.  I was mistaken when I told you it was like a really big binary number.  It's mostly true, but when the number overflows and you need to add another digit, it works a little different because of memcmp ordering.